tests/optimizers/test_scalar_update_functions.py

```diff
@@ -0,0 +1,148 @@
+        torch.manual_seed(42)
+        
+        # Create tensors
+        shape = (32, 16)
+        X = torch.randn(shape, device=device)
+        G = torch.randn(shape, device=device) * 0.01
+        M = torch.zeros(shape, device=device)
+        V = torch.zeros(shape, device=device)
+        
+        # Parameters
+        lr = torch.tensor(0.001)
+        beta1 = torch.tensor(0.9)
+        beta2 = torch.tensor(0.999)
+        weight_decay = torch.tensor(0.01)
+        epsilon = torch.tensor(1e-8)
+        step = torch.tensor(1)
+        
+        # Store original for comparison
+        X_orig = X.clone()
+        
+        # Call update function
+        try:
+            # The function might be compiled, which could fail in some environments
+            adamw_update(X, G, M, V, lr, beta1, beta2, weight_decay, epsilon, step)
+            
+            # Check that parameters were updated
+            assert not torch.allclose(X, X_orig), "Parameters were not updated"
+            
+            # Check momentum was updated
+            assert not torch.allclose(M, torch.zeros_like(M)), "Momentum was not updated"
+            
+            # Check variance was updated
+            assert not torch.allclose(V, torch.zeros_like(V)), "Variance was not updated"
+            
+        except Exception as e:
+            # If torch.compile fails, that's okay for testing
+            if "torch.compile" in str(e) or "dynamo" in str(e):
+                pytest.skip("torch.compile not available in this environment")
+            else:
+                raise
+    
+    def test_lion_update_function(self, device):
+        """Test lion_update function directly"""
+        torch.manual_seed(42)
+        
+        # Create tensors
+        shape = (32, 16)
+        X = torch.randn(shape, device=device)
+        G = torch.randn(shape, device=device) * 0.01
+        M = torch.zeros(shape, device=device)
+        
+        # Parameters
+        lr = torch.tensor(0.001)
+        beta1 = torch.tensor(0.9)
+        beta2 = torch.tensor(0.99)
+        weight_decay = torch.tensor(0.01)
+        
+        # Store original for comparison
+        X_orig = X.clone()
+        
+        # Call update function
+        try:
+            lion_update(X, G, M, lr, beta1, beta2, weight_decay)
+            
+            # Check that parameters were updated
+            assert not torch.allclose(X, X_orig), "Parameters were not updated"
+            
+            # Check momentum was updated
+            assert not torch.allclose(M, torch.zeros_like(M)), "Momentum was not updated"
+            
+        except Exception as e:
+            # If torch.compile fails, that's okay for testing
+            if "torch.compile" in str(e) or "dynamo" in str(e):
+                pytest.skip("torch.compile not available in this environment")
+            else:
+                raise
+    
+    def test_update_functions_with_weight_decay(self, device):
```

Comment:
There are some duplicate tests. Weight decay for AdamW and Lion are already tested in another file. Please look through all the test and remove any duplicates.